databuilder/databuilder/models/application.py

```diff
@@ -46,15 +54,54 @@ def __init__(self,
                  cluster: str = 'gold',
                  schema: str = '',
                  table_name: str = '',
-                 exec_date: str = '',
+                 application_type: str = 'Airflow',
                  ) -> None:
-        self.task = task_id
-
         # todo: need to modify this hack
         self.application_url = application_url_template.format(dag_id=dag_id)
-        self.database, self.cluster, self.schema, self.table = db_name, cluster, schema, table_name
-
+        self.database = db_name
+        self.cluster = cluster
+        self.schema = schema
+        self.table = table_name
         self.dag = dag_id
+        self.application_type = application_type
+        self.task = task_id
+
+        application_id_format = Application.APPLICATION_ID_FORMAT
+        application_key_format = Application.APPLICATION_KEY_FORMAT
+        application_description_format = Application.APPLICATION_DESCRIPTION_FORMAT
+
+        # The Application model was originally designed to only be compatible with Airflow
+        # If the type is Airflow we must use the hardcoded Airflow constants for backwards compatibility
+        if self.application_type == 'Airflow':
```

Comment:
just to be 200% sure can we do `if self.application_type.lower() == 'airflow'` ?

Reply:
Yup! Done ✅